### PR TITLE
feat: W5 Web Dashboard

### LINKED ISSUE #5: [Epic] W5: Web Dashboard
## Overview

Build the web dashboard for monitoring, managing, and approving agent work — the primary user interface for the CEO↔PM interaction model.

### Dashboard Foundation
- Next.js 15 App Router + Tailwind CSS
- API routes for all CRUD operations
- WebSocket for real-time updates
- Responsive layout with navigation

### Core Pages
- **Goals page**: Goal creation, progress tracking, epic/ticket hierarchy
- **Ticket Board**: Kanban with columns (Pending Approval → Backlog → In Progress → Verification → Done)
- **Heartbeat Report page**: Heartbeat logs, proposals, reverse proposals
- **Agent Management**: Agent list, status indicators, Soul/Identity editor
- **Direct Channel**: Chat UI with Team Lead message forwarding
- **Activity Log**: Real-time activity stream with filtering
- **Conventions Editor**: Edit CONVENTIONS.md / ARCHITECTURE.md / STYLE.md with change history

### Approval UI
- Pending Approval Kanban column with [Approve All] [Review Individually] [Reject]
- Reverse proposal approval cards
- Convention change proposal review

## Success Criteria
- [ ] Dashboard renders with all pages accessible
- [ ] Goals can be created and decomposed via UI
- [ ] Ticket Board shows real-time ticket state updates via WebSocket
- [ ] Approval flow works end-to-end in UI
- [ ] Direct Channel sends/receives messages with Team Lead
- [ ] Activity Log shows real-time agent actions
- [ ] Conventions can be edited and saved

## Dependencies
- #3 W3: Data Layer (API data source)
- #4 W4: Orchestrator (real-time status)

## Estimated Effort
~5 days (1 developer)

### PR DESCRIPTION
**Pull request by 12OneTwo12**
## Summary

Implements the Web Dashboard for monitoring, managing, and approving agent work — the primary UI for the CEO↔PM interaction model.

Closes #5

### Dashboard Foundation
- Next.js 15 App Router + Tailwind CSS 4
- 12 API routes (REST) for all CRUD operations
- SSE (Server-Sent Events) for real-time updates
- Responsive layout with sidebar navigation

### Core Pages
- **Goals page**: Goal creation, progress tracking, epic/ticket hierarchy
- **Ticket Board**: Kanban with 8 status columns (Pending Approval → Backlog → Assigned → In Progress → Verification → Done → Failed → Escalated)
- **Heartbeat Report page**: Heartbeat logs, proposals, reverse proposals
- **Agent Management**: Agent list, status indicators, 4-tab Soul/Identity editor
- **Direct Channel**: Chat UI with Team Lead message forwarding (SSE-powered)
- **Activity Log**: Real-time activity stream with agent/ticket filtering
- **Conventions Editor**: 3-tab editor (Conventions/Architecture/Style) with version tracking

### Approval UI
- Pending Approval Kanban column with Approve/Reject actions
- Reverse proposal approval cards
- Heartbeat proposal review

## Architecture
- Server-side DB access via `@phalanx/core` repositories (same-process import)
- EventBus → SSE bridge for real-time updates
- SWR for client-side data fetching
- TypeScript strict mode, all types from `@phalanx/core`

## Tests
- **600 tests passing** (66 files) — 521 core + 79 dashboard
- API route tests, EventBus tests, utility tests, edge-case tests

## Commits (9)
- 6 feature commits + 3 review round improvements

### CODE REVIEW

File: packages/dashboard/src/app/api/activity/route.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+export async function GET(request: NextRequest) {
+  const { searchParams } = request.nextUrl;
+  const agentId = searchParams.get('agentId');
+  const ticketId = searchParams.get('ticketId');
+  const limit = parseInt(searchParams.get('limit') ?? '100', 10);
+  const repo = getActivityLogRepository();
+
+  if (agentId) return jsonResponse(repo.findByAgentId(agentId));
+  if (ticketId) return jsonResponse(repo.findByTicketId(ticketId));
+  return jsonResponse(repo.findAll({ limit }));
```
[the remark this answers is not in the document]
Comment (12OneTwo12) [reply packages/dashboard/src/app/api/activity/route.ts]: Fixed in 285c5b8. Changed from precedence-based (`if agentId → else if ticketId`) to combinable filters — now fetches all logs via `findAll()` and applies both `agentId` and `ticketId` filters in-memory. Both params can be used simultaneously without silent drops.

File: packages/dashboard/src/app/api/agents/[id]/route.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  }
+
+  const repo = getAgentRepository();
+  try {
+    const updated = repo.update(id, sanitized);
+    if (!updated) return errorResponse('Agent not found', 404);
+    return jsonResponse(updated);
+  } catch (err) {
+    return errorResponse('Failed to update agent', 500);
+  }
```
[the remark this answers is not in the document]
Comment (12OneTwo12) [reply packages/dashboard/src/app/api/agents/[id]/route.ts:48]: Fixed in 285c5b8. Now extracts the actual error message from the caught exception and includes it in the 500 response: `Failed to update agent: <actual message>`. Also added `console.error` for server-side logging with the route path context.

File: packages/dashboard/src/app/api/agents/route.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+/** GET /api/agents — list all agents with optional role/status filter */
+export async function GET(request: NextRequest) {
+  const { searchParams } = request.nextUrl;
+  const role = searchParams.get('role') as NewAgent['role'] | null;
+  const status = searchParams.get('status') as NewAgent['status'] | null;
+  const repo = getAgentRepository();
+
+  if (role) return jsonResponse(repo.findByRole(role));
+  if (status) return jsonResponse(repo.findByStatus(status));
+  return jsonResponse(repo.findAll());
```
[the remark this answers is not in the document]
Comment (12OneTwo12) [reply packages/dashboard/src/app/api/agents/route.ts]: Fixed in 285c5b8. Changed from precedence-based (`if role → else if status`) to combinable filters — now uses `findAll()` with sequential `.filter()` calls. Both `role` and `status` can be provided simultaneously and will be intersected.

File: packages/dashboard/src/app/api/goals/[id]/route.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+}
+
+/** PATCH /api/goals/:id — update a goal */
+export async function PATCH(request: Request, { params }: RouteParams) {
+  const { id } = await params;
+  const body = await parseBody<{ description?: string; status?: string; progress?: number }>(request);
+  if (!body) return errorResponse('Invalid JSON body');
+
+  const repo = getGoalRepository();
+  const updated = repo.update(id, body);
```
[the remark this answers is not in the document]
Comment (12OneTwo12) [reply packages/dashboard/src/app/api/goals/[id]/route.ts]: Fixed in 285c5b8. Added `GOAL_UPDATABLE_FIELDS` whitelist (`description`, `status`, `progress`, `metadata`) consistent with the agents/tickets PATCH pattern. Unknown fields are now stripped before passing to `repo.update()`, and an empty sanitized body returns 400.

File: packages/dashboard/src/app/api/tickets/route.ts [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+export async function GET(request: NextRequest) {
+  try {
+    const { searchParams } = request.nextUrl;
+    const status = searchParams.get('status') as NewTicket['status'] | null;
+    const epicId = searchParams.get('epicId');
+    const repo = getTicketRepository();
+
+    if (status) return jsonResponse(repo.findByStatus(status));
+    if (epicId) return jsonResponse(repo.findByEpicId(epicId));
+    return jsonResponse(repo.findAll());
```
[the remark this answers is not in the document]
Comment (12OneTwo12) [reply packages/dashboard/src/app/api/tickets/route.ts]: Fixed in 285c5b8. Changed from precedence-based (`if status → else if epicId`) to combinable filters — now uses `findAll()` with sequential `.filter()` calls. Both `status` and `epicId` can be provided simultaneously and will be intersected.
